physicalplan/binary/scalar.go

```diff
@@ -81,7 +81,7 @@ func (o *scalarOperator) Next(ctx context.Context) ([]model.StepVector, error) {
 		step := o.pool.GetStepVector(vector.T)
 		for i := range vector.Samples {
 			lhs, rhs := o.getOperands(vector, i, o.scalar)
-			val := o.operation(lhs, rhs)
+			val, _ := o.operation(lhs, rhs)
```

Comment:
I cannot ignore the second return value. It is not compilable. 

```
../physicalplan/binary/scalar.go:84:11: assignment mismatch: 1 variable but o.operation returns 2 values
```

Reply:
Sorry, what I was saying is that we should also check if the second return value is true, and if not, probably we need to drop the resulting series.

Reply:
Scalar operation is always keep https://github.com/prometheus/prometheus/blob/main/promql/engine.go#L2182 while vector operation could drop samples somehow https://github.com/prometheus/prometheus/blob/main/promql/engine.go#L2215. That's why I skip it here but I can check if needed.

Reply:
> Let's add a test for something like sum(foo) > 10 to make sure we're not missing something.

This seems a bug. Working on this fix.

Reply:
I see the issue, so right now we go to `vectorOperator` if both sides are vectors. If one side is vector and another side is scaler we will still go to `scalarOperator`. This could be an issue since I need to check the two sides and decide the operation I need to use.

Reply:
I updated a version to fix this. But the code right now is super ugly. Probably better to refactor it a bit...